tests/checkout/tree.c

```diff
@@ -1551,8 +1566,12 @@ void test_checkout_tree__mode_change_is_force_updated(void)
 	cl_must_pass(p_chmod("testrepo/README", 0755));
 	cl_must_pass(git_index_add_bypath(index, "README"));
 
+	cl_git_pass(git_index_write(index));
 	assert_status_entrycount(g_repo, 1);
-	cl_git_pass(git_checkout_tree(g_repo, obj, &g_opts));
+
+	cl_git_pass(git_reset(g_repo, obj, GIT_RESET_HARD, NULL));
```

Comment:
Doesn't this change what we're actually testing? Previously, we've checked that `git_checkout_tree` resets the mode. Now we check whether `git_reset` resets the mode.